pontoon/base/models.py

```diff
@@ -1404,23 +1408,28 @@ def for_project_locale(self, project, locale, paths=None, status=None,
                 else:
                     query = q
 
-            if query:
-                entities = entities.with_status_counts(locale).filter(query)
-
         if time:
             if re.match('^[0-9]{12}-[0-9]{12}$', time):
                 try:
                     start, end = utils.parse_time_interval(time)
                 except ValueError:
                     raise ValueError(time)
 
-                entities = entities.between_time_interval(locale, start, end)
+                entities = entities.filter(Entity.objects.between_time_interval(locale, start, end))
 
             else:
                 raise ValueError(time)
 
         if author:
-            entities = entities.authored_by(locale, author)
+            q = Entity.objects.authored_by(locale, author)
+
+            if query:
+                query &= q
+            else:
+                query = q
+
```

Comment:
Replace this entire block with `entities = entities.filter(Entity.objects.authored_by(author))`. It will fix the same issue as with suggested & time range, but for authors.